test/Npgsql.Tests/SecurityTests.cs

```diff
@@ -276,33 +289,50 @@ public async Task Connect_with_only_non_ssl_allowed_user([Values] bool multiplex
     }
 
     [Test]
-    public void Connect_with_VerifyFull_and_callback_throws()
+    public void User_callback_is_invoked()
     {
         var csb = new NpgsqlConnectionStringBuilder(ConnectionString)
         {
-            SslMode = SslMode.VerifyFull
+            SslMode = SslMode.Require,
+            Pooling = false
+        };
+
+        using var connection = CreateConnection(csb.ToString());
+        connection.UserCertificateValidationCallback = (_, _, _, _) => false;
```

Comment:
Probably should test both cases (callback returns `true` and `false`) just to be sure.
Also, it should set some boolean variable to `true` and we should assert it's actually set.

Reply:
Good points.